### PR TITLE
dd: return UResult from uumain() function

### PR DESCRIPTION
**Pull request by jfinkels**
See #2464.

I updated one unit test so that it better matches the behavior of GNU dd in the case where there is no outfile.

### CODE REVIEW

File: src/uu/dd/src/datastructures.rs [hunk 1]
```diff
@@ -155,6 +157,12 @@ impl std::fmt::Display for InternalError {
 
 impl Error for InternalError {}
 
+impl UError for InternalError {
```
Comment (tertsdiepraam) [inline src/uu/dd/src/datastructures.rs:160]: If I remember correctly, the code of `1` is also the default implementation, so you could leave that out. However, it also makes sense to keep it, I guess.

**Review by tertsdiepraam — APPROVED** [review]
Looks good! Could you fix the `clippy` warning?

File: src/uu/dd/src/datastructures.rs (hunk 1 above)
Comment (jfinkels) [reply src/uu/dd/src/datastructures.rs:160]: Thanks, I'll replace this with `impl UError for InternalError {}` instead and keep that in mind for next time.

**Comment by jfinkels** [comment]
I have added a fixup commit making the suggested changes.

**Pull request merged**